app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -33,6 +33,23 @@ def fetch(hash)
     end
   end
 
+  # instead of storing records, only store ids
+  class IdTypedIndex < TypedIndex
+    def initialize(ids, find_key, model)
+      @key_attribute_types = find_key.map { |k| model.type_for_attribute(k) }
+
+      # find the records keys for the given records
+      records = model.where(model.primary_key => ids).pluck(*find_key, model.primary_key)
```

Comment:
hm, the queries with :id => [100k size array of ids] are usually not very effective :-(

Reply:
Yeah, I am aware, though couldn't think of a better way to do this at the time.

That said, commented on this below, and I think I can make this slightly more flexible if I do some of the ideas I had in https://github.com/ManageIQ/manageiq/pull/15989/files#r139717370, so we will see how that works out.

Reply:
Is it possible to keep these as a `select` / inner query instead of a `pluck`?

Reply:
@kbrock yes, that is what I was planning on doing.

The reason that I didn't originally is I was using this in two places:

* Where the normal `TypedIndex` was being used in `save_inventory_multi`, which is just basically doing a `association.all`, so a `.select` inner query would work.
* As part of my new method, `store_ids_for_record_ids`, which isn't something you can really do a select for since it is using the ids there are remaining from the `deletes` hash.

That said, as @Ladas pointed out [in his comment](https://github.com/ManageIQ/manageiq/pull/15989/files#r139717370), I don't really need to the second one since that method can just be done as part saving the record, and we can just update the ID at the same time (we think).

If not, we can just allow the initializer for `IdTypedIndex` to allow either a relation or an array, and then it can either create a inner query or `pluck` as needed.